### ISSUE TITLE
Question about training dataset for reproduction

### ISSUE THREAD

**Issue by XiaoXie658** [xslidi/CATMO#4]
Hello authors, thanks for releasing the CATMO code.

I am an undergraduate student reproducing your paper for my thesis. I tried to build the dataset following the paper, but still observe strong artifacts. I was wondering if the original training dataset (or part of it) could be shared for academic reproduction, or if there are any additional preprocessing details.

Any guidance would be greatly appreciated.

**Comment by xslidi** [comment]
Unfortunately, I am unable to share the original dataset, as it was not preserved. 
The dataset was collected from multiple sources, as described in the paper, with a significant portion originating from videos that were converted into static images for training purposes. The unpaired label images were primarily downloaded from Unsplash (unsplash.com). No additional preprocessing was applied beyond the methods provided by this library.

By the way, it is expected that unexpected artifacts may appear when this method is evaluated on datasets other than the Fairchild dataset used in our paper.

**Comment by XiaoXie658** [comment]
Thank you very much for your clarification.

During my dataset construction, I intentionally avoided using a large number of video-extracted frames, and mainly relied on standalone HDR images collected from different sources. Based on your explanation, this difference in data composition may indeed be one of the reasons for the artifacts I observe during evaluation.

I'm gonna reconstruct my dataset to see if the outcome gets better.

**Comment by XiaoXie658** [comment]
Thanks again for your previous reply.

I’d like to apologize if some of my questions sounded naive — I’m still relatively inexperienced with training stability issues in practice, so I’m learning a lot through trial and error during this reproduction.

I’m continuing to experiment with data filtering and cropping strategies, and I mainly wanted to check whether the training instability I’m observing is something you have also encountered in practice, rather than an obvious mistake on my side.

After rebuilding my dataset (mainly video frames, filtering extreme cases, and experimenting with cropping strategies), the loss curves and visual results do not show clear convergence, even after many epochs. I’m wondering if this behavior matches your experience. I would really appreciate your thoughts on a few points:

1. When preparing the HDR training data, did you apply any implicit filtering in practice (e.g., removing very dark frames, frames with limited local dynamic range, or manual visual inspection)?
2. Although random cropping is mentioned in the paper, did you ever observe instability related to random crops, especially on video frames or low-luminance regions?
3. In your experiments, how did you usually judge that the training had “converged”? Was it mainly based on visual quality, loss stabilization, or a rough number of epochs?

I completely understand if these details are not fully documented, but even a brief confirmation based on your experience would be very helpful for my reproduction work.

Since new year is comming, I don’t want to disturb you too much, Thank you very much for your time and for sharing your work, and I wish you a happy new year!

**Comment by xslidi** [comment]
I do not recall encountering any training instability, and I doubt it comes from the video/image ratio.

1. During data preparation, I manually removed duplicated images. However, I did not perform detailed visual inspection, as the HDR images cannot be accurately displayed on my monitor.

2. I am not entirely sure what you mean by “instability.” Are you referring to suboptimal performance on the Fairchild dataset?

3. The combination of visual quality and loss.

**Comment by XiaoXie658** [comment]
Thanks a lot for the clarification, that helps a lot.

By “instability” I mainly meant that during training, the loss curves do not seem to stabilize clearly and the visual outputs (e.g., halos or local artifacts) do not consistently improve over epochs, rather than suboptimal performance on the Fairchild dataset specifically.

Based on your explanation, I’ll continue experimenting on my side and see if the behavior improves. I may follow up later if I find anything more concrete.

Thanks again for your time and insights, and happy new year!

**Comment by XiaoXie658** [comment]
Sorry for bothering again. I recreated my training dataset and retrained the model multiple times, but I’m still encountering issues that prevent the model from behaving as expected.

I’m describing my setup in detail in case there is something incorrect or inconsistent that I may have overlooked.

For the LDR domain, I used the same image sources described in the paper, with random sampling. For the HDR domain, approximately one third of the images come from the video sources mentioned in the paper, and the remaining images are from PolyHaven. The images were selected randomly without any filtering based on dynamic range. I only removed duplicated images and randomly cropped all images to 512×512.

Previously, when training on a smaller dataset (In which each source image was randomly cropped multiple times to form multiple training samples) , I did not encounter severe issues. Although the final results were not as strong as those reported in the paper, the training behavior was generally reasonable: a clear adversarial relationship between G and D could be observed, and the NCE loss continued to decrease toward the later stages of training.

<img width="1538" height="822" alt="Image" src="https://github.com/user-attachments/assets/7ed12857-3409-48cd-bf6f-a45992ff4769" />

loss curve from small dataset training

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/b84dbe45-f1a7-4e55-aab1-0bd20e4b2469" />

epoch100_revise_B.png from small dataset training

In contrast, with the current larger dataset, the adversarial interaction between G and D is much less evident, and the NCE loss fails to show a clear downward trend in the later training stages., and the final visual quality of the generated images is mediocre.

<img width="1571" height="837" alt="Image" src="https://github.com/user-attachments/assets/fcd2209a-068b-4c40-a5cc-6c9a7cf23fe1" />

loss curve from this new dataset training

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/1f1e141a-0955-48e3-9a4c-9399a281b133" />

epoch100_revise_B.png from this new dataset training

At this point, I’m not sure what causes this behavior, nor whether it is directly responsible for the degraded results.

I followed the GitHub training instructions exactly, except that visualization was disabled:

python train.py --name CATMO --dataroot ./data/HDR --batch_size 2 --gpu_ids 0 \
--netG aunet --model cut --lambda_NCE 10 --nce_includes_all_negatives_from_minibatch \
--netD basic --spectral_norm


Environment:

Python: 3.13.11

PyTorch: 2.6.0+cu124

CUDA (PyTorch): 12.4

cuDNN: 90800

CUDA available: True

GPU: NVIDIA A10 (22 GiB / 23 GiB used)

Driver: 570.133.20, System CUDA: 12.8

OS: Ubuntu 5.15.0-161-generic x86_64

I’m still trying to reproduce the results but haven’t resolved this issue yet. I would like to ask whether the lack of a clear adversarial relationship between G and D could be caused by the increased dataset diversity, or whether there is something else in the training setup that I might be overlooking.I am honestly a bit stuck on how to proceed with the reproduction, and would really appreciate any guidance on what I might be missing.

**Comment by xslidi** [comment]
There is a pretrained latest_g provided in this repo, maybe you can load the weight and test its performance on your training dataset.

**Comment by XiaoXie658** [comment]
(epoch: 1, iters: 100, time: 0.359, data: 0.141) D: 0.253 G: 35.166 NCE: 34.548 style: 0.074

I do not understand this.
I simply pasted the  latest_g into my CATMO checkpoint then run train.py with

python train.py --name CATMO --dataroot ./data/HDR --batch_size 2 --gpu_ids 0 --netG aunet --model cut --lambda_NCE 10 --nce_includes_all_negatives_from_minibatch --netD basic --spectral_norm --display_id 0 --epoch latest

Did i do it wrong?

**Comment by xslidi** [comment]
Please only test it but not train it.

**Comment by XiaoXie658** [comment]
Sorry about that mistake
It warned me "smaller image, not resized!" but i still got output.
A few of them have black dots around high light like this one, but most look visually great.

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/74bd4175-eb1d-435d-b4f0-8ef546e65b19" />

**Comment by xslidi** [comment]
It sounds like there may be an issue with the training process. Normally, after convergence, the NCE loss should settle to a relatively low value, but yours remains unusually high—even in the first chart you shared—which suggests the training may have failed.

There are many possible reasons this can happen, and since this is a project I worked on a long time ago, I’m not able to provide much specific troubleshooting. For now, I’d recommend ignoring my previous hyperparameter settings and focusing on getting the model to reliably converge first.

As for the training dataset, I doubt my handcrafted set was some kind of “magic recipe.” I don’t remember spending much time on it, and I don’t think it was designed with particular care.

**Comment by XiaoXie658** [comment]
Thank you very much for the detailed explanation and for taking the time to look into this, I really appreciate it.

Your comments about the NCE loss and convergence are very helpful and confirm some of my observations. I’ll focus on simplifying the setup and getting stable convergence first, rather than strictly following the original hyperparameter settings.

Thanks again for your patience and guidance.

**Comment by XiaoXie658** [comment]
I’ve figured it out.
The reason my previous training attempts failed is that, in the current code, the NCE loss is computed using revise_B instead of fake_B.
